### ISSUE TITLE
[java] CloneMethodMustImplementCloneable gives false positives

### ISSUE THREAD

**Issue by Noah0120** [pmd/pmd#3441]
**Affects PMD Version:** 
6.36.0
**Rule:**
CloneMethodMustImplementCloneable
Please provide the rule name and a link to the rule documentation:
https://pmd.github.io/latest/pmd_rules_java_errorprone.html#clonemethodmustimplementcloneable

**Description:**

**Code Sample demonstrating the issue:**

```

```

**Expected outcome:**

**Running PMD through:** *[CLI]*

**Comment by Noah0120** [comment]
I found it is invalid. Thank you.
